src/main/kotlin/lotto/LottoMachine.kt

```diff
@@ -0,0 +1,34 @@
+package lotto
+
+class LottoMachine(val purchaseAmount: Int, val tickets: MutableList<Lotto> = emptyList<Lotto>().toMutableList()) {
+    private val numberList = (Lotto.Companion.MIN..Lotto.Companion.MAX)
+    private var change = 0
+    private var ticketCount = 0
+
+    init {
+        require(purchaseAmount in MIN..MAX) {
+            throw IllegalArgumentException()
+        }
+        change = purchaseAmount % TICKET_PRICE
+        ticketCount = (purchaseAmount - change) / TICKET_PRICE
+        generateTickets()
+    }
+
+    fun showChange() = change
+
+    fun generateTickets() {
+        repeat(ticketCount) {
+            tickets.add(Lotto(generateNumbers()))
+        }
+    }
+
+    private fun generateNumbers(): List<Int> {
+        return numberList.shuffled().subList(0, Lotto.LOTTO_SIZE)
+    }
```

Comment:
how about this? kotlin magic-
```suggestion
    private fun generateNumbers(): List<Int> {
        return numberList
            .shuffled()
            .take(Lotto.LOTTO_SIZE)
    }
```